get_experiment_data.R

```diff
@@ -187,17 +194,28 @@ if(opt$get_marker_genes & !is.na(opt$number_of_clusters)){
     multiple_markers = TRUE
 }
 
-names = c("sdrf.txt", "condensed-sdrf.tsv", "idf.txt", markers) 
+names = c("sdrf.txt", "condensed-sdrf.tsv", "idf.txt", markers, 
+          paste("'https://www.ebi.ac.uk/gxa/sc/experiment", acc, 
+          "download?fileType=experiment-design&accessKey='",sep="/"))
+
 for(idx in seq_along(non_expr_files)){
-    if(non_expr_files[idx]){
-        url = paste(url_prefix, names[idx], sep=".")
-        i = paste(output_dir, basename(url), sep="/")
-        system(paste("wget", url, "-P", output_dir))
-        if(!file.exists(i)) stop(paste("File", i, "does not exist"))
-        # do not rename if multiple marker files downloaded
-        if(!opt$use_full_names & !(idx==4 & multiple_markers)){
-            o = paste(output_dir, names[idx], sep="/")
-            file.rename(i, o)
+    metadata_file = non_expr_files[idx]
+    if(metadata_file){
+        if(idx == 5){
+            url = names[idx]
+            system(paste("wget", url, "-P", output_dir))
+            file.rename(paste(output_dir, "download?fileType=experiment-design&accessKey=",sep="/"),
```

Comment:
instead of the rename, you could outright give wget the name that you want for the output file

Reply:
@pcm32 I wanted to do that initially, but weirdly you can't use both -P and -O tags with wget